src/fsm/mod.rs

```diff
@@ -0,0 +1,38 @@
+use core::cell::Cell;
+
+#[cfg(test)]
+mod test;
+
+pub enum FsmSignal<UserSignal> {
+    Entry,
+    Exit,
+    User(UserSignal),
+}
+
+pub type FsmState<Type, UserSignal> = fn(obj: &Type, signal: FsmSignal<UserSignal>) -> ();
```

Comment:
is there a reason that we use `fn` instead of one of the `Fn` traits? I don't immediately see a reason to disallow closures

Reply:
Though if we want it to be a Trait Object, it could still work 🤔.